### PR TITLE
Users/rahalan/add vnet child resources

### PR DESCRIPTION
**Pull request by rahalan**
# Change

Change vNet module structure to have vnet peering as a child resource

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update (Wiki)

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (readme)
- [x] I did format my code

### CODE REVIEW

File: .github/workflows/ms.network.virtualnetworks.virtualnetworkpeerings.yml [hunk 1]
```diff
@@ -21,12 +21,12 @@ on:
     paths:
       - '.github/actions/templates/**'
       - '.github/workflows/ms.network.virtualnetworks.virtualnetworkpeerings.yml'
```
Comment (MrMCake) [inline .github/workflows/ms.network.virtualnetworks.virtualnetworkpeerings.yml:23]: - Please remove this file. We don't want pipelines for child resources
- Also remove the `.parameters` folder from the child resource. The contained test-parameters for the child should move to a top-level parameter file

File: arm/Microsoft.Network/virtualNetworks/deploy.bicep [hunk 2]
```diff
@@ -135,7 +152,7 @@ resource virtualNetwork_lock 'Microsoft.Authorization/locks@2016-09-01' = if (lo
 }
 
 resource appServiceEnvironment_diagnosticSettings 'Microsoft.Insights/diagnosticSettings@2017-05-01-preview' = if (!empty(diagnosticStorageAccountId) || !empty(workspaceId) || !empty(eventHubAuthorizationRuleId) || !empty(eventHubName)) {
-  name: '${vNetName}-diagnosticSettings'
+  name: '${name}-diagnosticSettings'
```
Comment (MrMCake) [inline arm/Microsoft.Network/virtualNetworks/deploy.bicep:155]: update please to `name: '${virtualNetwork.name}-diagnosticSettings'`

File: arm/Microsoft.Network/virtualNetworks/readme.md [hunk 3]
```diff
@@ -10,6 +10,7 @@ This template deploys a Virtual Network (vNet) with 2 optional Subnets.
 | `Microsoft.Authorization/roleAssignments` | 2020-04-01-preview |
```
Comment (MrMCake) [inline arm/Microsoft.Network/virtualNetworks/readme.md:10]: Regenerate with function `utilities\tools\Set-ModuleReadMe.ps1`. You may need to merge the latest main back into your branch to get the latest, fixed version

File: arm/Microsoft.Network/virtualNetworks/virtualNetworkPeerings/deploy.bicep [hunk 4]
```diff
@@ -31,7 +31,7 @@ module pid_cuaId '.bicep/nested_cuaId.bicep' = if (!empty(cuaId)) {
 }
 
 resource virtualNetworkPeering 'Microsoft.Network/virtualNetworks/virtualNetworkPeerings@2021-02-01' = {
-  name: '${localVnetName}/${peeringName}'
+  name: '${localVnetName}/${name}'
```
Comment (MrMCake) [inline arm/Microsoft.Network/virtualNetworks/virtualNetworkPeerings/deploy.bicep:34]: Please add `@description('...')` for outputs and regenerate the docs like described in the other readme comment

File: .github/workflows/ms.network.virtualnetworks.virtualnetworkpeerings.yml (hunk 1 above)
Comment (rahalan) [reply .github/workflows/ms.network.virtualnetworks.virtualnetworkpeerings.yml:23]: done

File: arm/Microsoft.Network/virtualNetworks/deploy.bicep (hunk 2 above)
Comment (rahalan) [reply arm/Microsoft.Network/virtualNetworks/deploy.bicep:155]: done

File: arm/Microsoft.Network/virtualNetworks/readme.md (hunk 3 above)
Comment (rahalan) [reply arm/Microsoft.Network/virtualNetworks/readme.md:10]: Already did it. Thanks to Marius

File: arm/Microsoft.Network/virtualNetworks/virtualNetworkPeerings/deploy.bicep (hunk 4 above)
Comment (rahalan) [reply arm/Microsoft.Network/virtualNetworks/virtualNetworkPeerings/deploy.bicep:34]: done

**Review by MariusStorhaug — APPROVED** [review]
All comments seem to be fixed. Approved.

**Pull request merged**